src/components/csvIntegration.js

```diff
@@ -0,0 +1,249 @@
+    this.setState({
+      data: {
+        ...this.state.data,
+        [name]: file,
+        [link]: value
+      }
+    });
+    event.target.value = "";
+  };
+
+  handleDelete = name => {
+    let link = name + "Link";
+    this.setState({
+      data: {
+        ...this.state.data,
+        [name]: null,
+        [link]: null
+      }
+    });
+  };
+
+  handleChange = (name, value) => {
+    this.setState({
+      data: {
+        ...this.state.data,
+        [value.name]: value.value
+      }
+    });
+  };
+
+  handleErrors = error_dict => {
+    let dict = error_dict;
+    let errors = [];
+    for (let key in dict) {
+      for (let index in dict[key]) {
+        errors.push(startCase(key) + ": " + dict[key][index]);
+      }
+    }
+    this.setState({ errors: errors });
+  };
+
+  handleSubmit = () => {
+    this.setState({ loading: true });
+    if (this.state.data.fileLink == null) {
+      this.setState({ 
+        errors: ["You need to upload file before submitting."],
+        loading: false
+      })
+    } else {
+        let data = new FormData();
+        data.append("model", this.state.modelName)
+        for (let prop in this.state.data) {
+          data.append(snakeCase(prop), this.state.data[prop])
+        }
+        axios({
+            method: "post",
+            url: urlWriteAppendMultipleObjects(),
+            data: data,
+            headers: headers
+        }).then(response => {
+          this.setState({ loading: false });
+          this.props.handleUpdate(response.data, this.props.componentName);
+          this.props.handleCsvHide(this.props.componentName);
+        }).catch(error => {
+            if (error.response.status === 400) {
+              this.setState({ loading: false });
+              this.handleErrors(error.response.data);
+            }
+        });
+      }
+  }
+
+  makeInstructions = affordances => {
+    let instructions = [];
+    for (let index in affordances) {
+      for (let [key, value] of Object.entries(affordances[index])) {
+        instructions.push(`${key}: ${value}`)
+      }
+    }
+    return instructions;
+  }
+
+  render() {
+    const { componentName, affordances, appDetails } = this.props;
+    const { modelName, data, errors, loading } = this.state;
+    let instructions = this.makeInstructions(affordances);
+    let FileComponent = FieldMap['file_field']
+    return (
+      <Segment basic styleName="style.csvMinWidth">
+        <Segment attached="top" styleName="style.headingBox">
+          <h3 styleName="style.heading">Add {modelName}s via File</h3>
+          <Icon
+            color="grey"
+            name="delete"
+            onClick={() => this.props.handleCsvHide(componentName)}
+          />
+        </Segment>
+        <Segment attached styleName="style.formStyle">
+          <ErrorTransition errors={errors} />
+          <Message
+            info
+            header="Column headers has to satisfy below shown order and properties."
```

Comment:
- ~has~ have
- the below

I think we should rephrase it as: "Column headers have to satisfy the below-stated constraints."